src/Common/Core/MorganStanley.ComposeUI.Common.Interfaces/ISyncContext.cs

```diff
@@ -0,0 +1,34 @@
+﻿/// ********************************************************************************************************
+///
+/// Morgan Stanley makes this available to you under the Apache License, Version 2.0 (the "License").
+/// You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0.
+/// See the NOTICE file distributed with this work for additional information regarding copyright ownership.
+/// Unless required by applicable law or agreed to in writing, software distributed under the License
+/// is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+/// See the License for the specific language governing permissions and limitations under the License.
+/// 
+/// ********************************************************************************************************
+
+namespace MorganStanley.ComposeUI.Common.Interfaces
+{
+    /// <summary>
+    /// Interface for marshalling action and its arguments to an STA thread (in which the SyncContext object had 
+    /// been created)
+    /// </summary>
+    public interface ISyncContext
```

Comment:
What about InvokeAsync for awaitable work?

`

Reply:
I can definitely add the method, though so far I did not need it.

Reply:
let us do it in a different jira - this is just a code skeleton, no need to overdo.